"azurerm_log_analytics_workspace.main" error: Plugin did not respond

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.5.7

### AzureRM Provider Version

3.71.0

### Affected Resource(s)/Data Source(s)

azurerm_app_configuration

### Terraform Configuration Files

```hcl
# module.app_insights.azurerm_application_insights.main will be created
  + resource "azurerm_application_insights" "main" {
      + app_id                                = (known after apply)
      + application_type                      = "other"
      + connection_string                     = (sensitive value)
      + daily_data_cap_in_gb                  = (known after apply)
      + daily_data_cap_notifications_disabled = (known after apply)
      + disable_ip_masking                    = false
      + force_customer_storage_for_profiler   = false
      + id                                    = (known after apply)
      + instrumentation_key                   = (sensitive value)
      + internet_ingestion_enabled            = true
      + internet_query_enabled                = true
      + local_authentication_disabled         = false
      + location                              = "eastus"
      + name                                  = "appi-product-sbx"
      + resource_group_name                   = "rg-product-sbx"
      + retention_in_days                     = 90
      + sampling_percentage                   = 100
      + tags                                  = {
          + "ENV" = "test"
        }
      + workspace_id                          = (known after apply)
    }

  # module.app_insights.azurerm_log_analytics_workspace.main will be created
  + resource "azurerm_log_analytics_workspace" "main" {
      + allow_resource_only_permissions = true
      + daily_quota_gb                  = -1
      + id                              = (known after apply)
      + internet_ingestion_enabled      = true
      + internet_query_enabled          = true
      + local_authentication_disabled   = false
      + location                        = "eastus"
      + name                            = "appi-prodcut-sbx-ws"
      + primary_shared_key              = (sensitive value)
      + resource_group_name             = "rg-product-sbx"
      + retention_in_days               = 90
      + secondary_shared_key            = (sensitive value)
      + sku                             = "PerGB2018"
      + tags                            = {
          + "ENV" = "test"
        }
      + workspace_id                    = (known after apply)
    }
```


### Debug Output/Panic Output

```shell
module.app_configurations.azurerm_app_configuration.main: Creating...
module.app_insights.azurerm_log_analytics_workspace.main: Creating...
╷
│ Error: Plugin did not respond
│
│   with module.app_configurations.azurerm_app_configuration.main,
│   on modules\az-appConfiguration\main.tf line 2, in resource "azurerm_app_configuration" "main":
│    2: resource "azurerm_app_configuration" "main" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵
╷
│ Error: Plugin did not respond
│
│   with module.app_insights.azurerm_log_analytics_workspace.main,
│   on modules\az-appInsights\main.tf line 2, in resource "azurerm_log_analytics_workspace" "main":
│    2: resource "azurerm_log_analytics_workspace" "main" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.71.0_x5.exe plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 135 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration.expandAppConfigurationEncryption(...)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/app_configuration_resource.go:654
github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration.resourceAppConfigurationCreate(0x17105d10, {0x5f3bf80, 0x15f04240})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/appconfiguration/app_configuration_resource.go:287 +0x1578
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x156aafc0, {0x7162624, 0x15c149c0}, 0x17105d10, {0x5f3bf80, 0x15f04240})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:695 +0x166
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x156aafc0, {0x7162624, 0x15c149c0}, 0x1710ee70, 0x17105c20, {0x5f3bf80, 0x15f04240})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:837 +0xa98
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x16308500, {0x7162624, 0x15c14960}, 0x16956cf0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xfa4
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x159dad70, {0x7162624, 0x15c14660}, 0x15a81a00)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x583
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x66ae9e0, 0x159dad70}, {0x7162624, 0x15c14660}, 0x15a819c0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x18a
google.golang.org/grpc.(*Server).processUnaryRPC(0x154a0ea0, {0x716b224, 0x1642c000}, 0x171320a0, 0x160e2348, 0xb8f1a20, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xe08
google.golang.org/grpc.(*Server).handleStream(0x154a0ea0, {0x716b224, 0x1642c000}, 0x171320a0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0xa1e
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0xb1
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x25e

Error: The terraform-provider-azurerm_v3.71.0_x5.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

In first time apply, It creates the azurerm workspace ID, But it fails with Plugin didnt responded. 
In second apply it fails with resource already exists. For App configuration resource its same plugin didnt respond.


### Actual Behaviour

_No response_

### Steps to Reproduce

Trying to create 3 resources with minimal configs 
azurerm_log_analytics_workspace
azurerm_application_insights
azurerm_app_configuration

Its giving error Plugin did not respond in first apply. 



### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appconfiguration/app_configuration_resource.go
+++ b/internal/services/appconfiguration/app_configuration_resource.go
@@ -647,12 +647,7 @@ type flattenedAccessKeys struct {
 }
 
 func expandAppConfigurationEncryption(input []interface{}) *configurationstores.EncryptionProperties {
-	if len(input) == 0 {
-		return nil
-	}
-
-	raw := input[0]
-	if raw == nil {
+	if len(input) == 0 || input[0] == nil {
 		return nil
 	}
 

--- a/internal/services/appconfiguration/app_configuration_resource.go
+++ b/internal/services/appconfiguration/app_configuration_resource.go
@@ -651,7 +651,13 @@ func expandAppConfigurationEncryption(input []interface{}) *configurationstores.
 		return nil
 	}
 
+	raw := input[0]
+	if raw == nil {
+		return nil
+	}
+
 	encryptionParam := input[0].(map[string]interface{})
+
 	result := &configurationstores.EncryptionProperties{
 		KeyVaultProperties: &configurationstores.KeyVaultProperties{},
 	}

